packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -83,6 +88,36 @@ export function cloudflare<
 				}
 			}
 
+			const esmResolveId = createIdResolver(viteConfig, {});
+
+			// for `require` calls we want a resolver that prioritized node/cjs modules
+			const cjsResolveId = createIdResolver(viteConfig, {
+				conditions: ['node'],
+				mainFields: ['main'],
+				webCompatible: false,
+				isRequire: true,
+				extensions: ['.cjs', '.cts', '.js', '.ts', '.jsx', '.tsx', '.json'],
+			});
+
+			const resolveId: ResolveIdFunction = (
+				id,
+				importer,
+				{ resolveMethod } = {
+					resolveMethod: 'import',
+				},
+			) => {
+				const resolveIdFn =
+					resolveMethod === 'import' ? esmResolveId : cjsResolveId;
+
+				const firstWorkerName = Object.keys(pluginConfig.workers)[0]!;
```

Comment:
Let's inlcude a comment here so we know this is temporary.

```suggestion
        // TODO: investigate making `resolveId` function unique for each worker
				const firstWorkerName = Object.keys(pluginConfig.workers)[0]!;
```

Reply:
what about this? 🙂 
```suggestion
				// TODO: we only have a single module resolution strategy shared across all workers
				//       (generated using the first worker's dev environment)
				//       we should investigate and ideally have potential different resolutions per worker
				//       see: https://github.com/flarelabs-net/vite-plugin-cloudflare/issues/19
```